src/sources/azure_blob/queue.rs

```diff
@@ -160,57 +161,69 @@ async fn proccess_event_grid_message(
         );
         return None;
     }
-    let parts = body.subject.split("/").collect::<Vec<_>>();
-    if parts.len() != 7 {
-        warn!("Ignoring event because of wrong subject format");
-        return None;
-    }
-    let container = parts[4];
-    if container != container_client.container_name() {
-        // This shouldn't happen if everything is configured properly.
-        return None;
-    }
-    let blob = parts[6];
-    info!(
-        "New blob created in container '{}': '{}'",
-        &container, &blob
-    );
+    match parse_subject(body.subject) {
+        Some((container, blob)) => {
+            if container != container_client.container_name() {
+                trace!("Container name does not match. Skipping.");
+                return None;
+            }
+            trace!(
+                "New blob created in container '{}': '{}'",
+                &container,
+                &blob
+            );
+            let blob_client = container_client.blob_client(blob);
+            let mut result: Vec<u8> = vec![];
+            let mut stream = blob_client.get().into_stream();
+            while let Some(value) = stream.next().await {
+                let mut body = value.unwrap().data;
+                while let Some(value) = body.next().await {
+                    let value = value.expect("Failed to read body chunk");
+                    result.extend(&value);
+                }
+            }
+
+            let reader = Cursor::new(result);
+            let buffered = BufReader::new(reader);
+            let queue_client_copy = queue_client.clone();
+            let bytes_received_copy = bytes_received.clone();
 
-    let blob_client = container_client.blob_client(blob);
-    let mut result: Vec<u8> = vec![];
-    let mut stream = blob_client.get().into_stream();
-    while let Some(value) = stream.next().await {
-        let mut body = value.unwrap().data;
-        while let Some(value) = body.next().await {
-            let value = value.expect("Failed to read body chunk");
-            result.extend(&value);
+            Some(BlobPack {
+                row_stream: Box::pin(stream! {
+                    for line in buffered.lines() {
+                        let line = line.map(|line| line.as_bytes().to_vec());
+                        let line = line.expect("ASDF");
+                        bytes_received_copy.emit(ByteSize(line.len()));
+                        yield line;
+                    }
+                }),
+                success_handler: Box::new(|| {
+                    Box::pin(async move {
+                        queue_client_copy
+                            .pop_receipt_client(message)
+                            .delete()
+                            .await
+                            .expect("Failed removing messages from queue");
+                    })
+                }),
+            })
+        }
+        None => {
+            trace!("Failed parsing subject. Skipping.");
```

Comment:
This should be either an error or a warning. It's not a sunny day path.

Reply:
Will change to warn.